smtp/smtpc.h

```diff
@@ -116,6 +116,7 @@ struct smtpc_request {
 	int status;
 	/** Error message. */
 	const char *reason;
+	char *error;
```

Comment:
Why we cannot reuse `reason` for our purposes?

Reply:
The signautre of function `CURLcode curl_easy_setopt(CURL *handle, CURLOPT_ERRORBUFFER, char *buf);` accepts a non-constant char buffer. The reference is [here](https://curl.se/libcurl/c/CURLOPT_ERRORBUFFER.html).